src/main/java/bisq/asset/AddressValidationResult.java

```diff
@@ -52,11 +52,15 @@ public static AddressValidationResult validAddress() {
     }
 
     public static AddressValidationResult invalidStructure() {
-        return new AddressValidationResult(false, "", "validation.altcoin.wrongStructure");
+        return invalidAddress("", "validation.altcoin.wrongStructure");
     }
 
     public static AddressValidationResult invalidAddress(String cause) {
-        return new AddressValidationResult(false, cause, "validation.altcoin.invalidAddress");
+        return invalidAddress(cause, "validation.altcoin.invalidAddress");
+    }
+
+    public static AddressValidationResult invalidAddress(String cause, String i18nKey) {
+        return new AddressValidationResult(false, cause, i18nKey);
     }
```

Comment:
I actually want to avoid exposing the `i18nKey` as much as possible. I knew this would come up with the special `zAddressesNotSupported` key, and it's fine if you leave this here for now, but in general, I want to get internationalization stuff out of the picture at this low level. It requires some more thinking about exactly how to do this, though, so again, no worries, and you can just leave this for now.

Reply:
Should I remove the translation from properties files next time I hit something like this?

Reply:
No, that's the thing. We want to perfectly preserve the i18n behavior that's in place right now, but (ideally) remove any awareness of these keys from the low-level `bisq.asset` package. So you can leave everything as-is. It is definitely important for the moment, that these i18n keys do continue to get propagated up to the desktop layer in the meantime, though.